server/task-json/src/main/java/org/apache/james/server/task/json/TaskDeserializer.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.james.server.task.json;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.james.task.Task;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class TaskDeserializer {
+
+    public interface Factory {
```

Comment:
if not thing special happened inside the Factory except using Jackson to deserialize a JsonNode into a right Task instance. I think you don't need that Factory and update the registry type to 
```java
private final Map<String, Class<? extends Task>> registry
```
 is enough

And I dont like the name of `Factory`, looks like using different creation patterns in one class. How about the name `Deserializer` ?

Reply:
Deserializer is going from binary or string format to datastructures.
Here we go from Json objects to Domain objects. So it can be either a conversion or a creation (so Converter or Factory for type names).
I still think Factory sounds good.